app/client/components/billing/invoicesTable.tsx

```diff
@@ -0,0 +1,326 @@
+import { css } from "@emotion/core";
+import { palette, space } from "@guardian/src-foundations";
+import { headline, textSans } from "@guardian/src-foundations/typography";
+import moment from "moment";
+import React, { useState } from "react";
+import { DATE_INPUT_FORMAT, formatDateStr } from "../../../shared/dates";
+import { InvoiceDataApiItem } from "../../../shared/productResponse";
+import { maxWidth, minWidth } from "../../styles/breakpoints";
+import { Pagination } from "../pagination";
+import { CardDisplay } from "../payment/cardDisplay";
+import { DirectDebitDisplay } from "../payment/directDebitDisplay";
+import { PayPalDisplay } from "../payment/paypalDisplay";
+import { DownloadIcon } from "../svgs/downloadIcon";
+import { InvoiceTableYearSelect } from "./invoiceTableYearSelect";
+
+const invoicePaymentMethods = {
+  CARD: "card",
+  DIRECT_DEBIT: "directdebit",
+  PAYPAL: "paypal"
+};
+
+interface InvoiceInfo extends InvoiceDataApiItem {
+  product?: string;
+  currencyISO: string;
+  currency: string;
+}
+
+interface InvoicesTableProps {
+  resultsPerPage: number;
+  invoiceData: InvoiceInfo[];
+}
+
+export const InvoicesTable = (props: InvoicesTableProps) => {
+  const initialPage = 1;
+
+  const [currentPage, setCurrentPage] = useState<number>(initialPage);
+
+  const tableHeadings = ["Date", "Payment method", "Price", ""];
+  const invoiceYears = [
```

Comment:
Does this rely on invoiceData being sorted by `invoice.date`?

Reply:
Hey, yep invoiceData is sorted by date before being passed through to the invoicesTable component as a prop.

I can see what you are saying, the component will work without the invoiceYears being sorted, you may just get a list of years not in chronological order.

Do you think it is worth explicitly sorting the years in this component, or leave it knowing that the data has already been sorted elsewhere?

Reply:
It's a tiny change to run `.sort()` on the end of the array assignment,  so possibly safer to do that than relying on Set() not reordering or the API changing.